db/statistics.go

```diff
@@ -1130,11 +1130,26 @@ func markColumnExported(day uint64, column string) error {
 	return nil
 }
 
-func GetValidatorIncomeHistoryChart(validatorIndices []uint64, currency string, lastFinalizedEpoch uint64) ([]*types.ChartDataPoint, error) {
-	incomeHistory, err := GetValidatorIncomeHistory(validatorIndices, 0, 0, lastFinalizedEpoch)
-	if err != nil {
-		return nil, err
+func GetValidatorIncomeHistoryChart(validatorIndices []uint64, currency string, lastFinalizedEpoch uint64, days uint64) ([]*types.ChartDataPoint, error) {
+	var incomeHistory []types.ValidatorIncomeHistory
+	var err error
+	if days == 0 {
+		incomeHistory, err = GetValidatorIncomeHistory(validatorIndices, 0, 0, lastFinalizedEpoch)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		tNow := time.Now()
+		lastDay := time.Date(tNow.Year(), tNow.Month(), tNow.Day(), 0, 0, 0, 0, time.UTC)
+		firstDay := lastDay.AddDate(0, -1, 0)
+		lowerBound := utils.TimeToDay(uint64(firstDay.Unix()))
+		upperBound := utils.TimeToDay(uint64(lastDay.Unix()))
+		incomeHistory, err = GetValidatorIncomeHistory(validatorIndices, lowerBound, upperBound, lastFinalizedEpoch)
```

Comment:
It looks like these changes introduce an undesired behavior change:
- Navigate to any validator, for example https://prater.beaconcha.in/validator/320764 on production
- Navigate to the same validator with your local branch

Production will show Consensus Income for the current day while your branch will not:

**Production:**
![image](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/114066135/59194ffd-4d61-42eb-b6a9-aa4963ba1a5b)

**Your branch:**
![image](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/114066135/b90a21fb-56e4-44de-8c6b-87001bcce895)